### DISCUSSION TITLE
launch.json not used when debugging from the bottom toolbar or from the cmake target context menu ?

### DISCUSSION THREAD

**Discussion by neundorf** [microsoft/vscode-cmake-tools#3188] [General]
I have created a launch.json, and VS Code uses it when I start debugging via the "Run and debug" tab, there I can select the configuration I want and it is used properly.
But when I start debugging via the debug button in the bottom status bar, or via the "Debug" entry in the context menu of a target in the cmake tab, this launch configuration is not used. I tried setting the working debug configuration via a "default" entry, but this also didn't work.
How can I make this work ?

**Comment by neundorf** [comment]
Below is my launch.json. Are there other settings which influence this ? Maybe in the cmake tools or somewhere ?
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "(msvc) Launch",
            "type": "cppvsdbg",
            "request": "launch",
            // Resolved by CMake Tools:
            "program": "${command:cmake.launchTargetPath}",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "envFile": "${workspaceFolder}/env.txt",
            "visualizerFile": "${workspaceFolder}/qt5.natvis.xml"
        }

    ],
    "default": "(msvc) Launch"
}
```

**Reply by bobbrow** [reply]
The buttons added by the CMake Tools extension are not part of the default VS Code workflow and are a way for folks to get a "quick debugging" experience without configuring `launch.json`.  You can hide the status bar button with the `cmake.statusbar.advanced` setting.
